django_ipam/tests/base/test_models.py

```diff
@@ -169,3 +169,27 @@ def test_retrieves_ipv6_ipnetwork_type(self):
         instance = self._create_subnet(subnet='2001:db8::0/32')
         instance = self.subnet_model.objects.get(pk=instance.pk)
         self.assertIsInstance(instance.subnet, IPv6Network)
+
+    def test_incompatible_ipadresses(self):
+        failed = True
+        instance = self._create_subnet(subnet='10.1.2.0/24')
+        try:
+            self._create_subnet(subnet='2001:db8::0/32', master_subnet=instance)
+        except TypeError as err:
+            self.assertEqual(str(err), '2001:db8::/32 and 10.1.2.0/24 are not of the same version')
+            failed = False
+        if failed:
```

Comment:
can't you do:

```python
def test_incompatible_ipadresses(self):
    instance = self._create_subnet(subnet='10.1.2.0/24')
    try:
        self._create_subnet(subnet='2001:db8::0/32', master_subnet=instance)
    except TypeError as err:
        self.assertEqual(str(err), '2001:db8::/32 and 10.1.2.0/24 are not of the same version')
    else:
        self.fail('TypeError not raised')
```

I usually use this form in the other openwisp modules, which is simpler and should be easy to read. What do you think?

Reply:
All other tests in the repository are done in this manner, hence for consistency i kept it in the same way. 
But indeed, this is not the best way to do it, so i think i should change all the test cases to follow rest of the OpenWISP modules! :smile: